DebugMenu/DebugMenu/Wrappers/DebugToggle.swift

```diff
@@ -9,78 +9,120 @@ import SwiftUI
 
     public let displayTitle: String
-    public let defaultValue: Value
-    public let storage: UserDefaults
-    public let key: String?
+    public let defaultValue: Bool
+    public var publisher: PassthroughSubject<Bool, Never>
+    public let didSet: ((Bool) -> Void)?
 
-    private var value: Value
+    @ObservedObject public private(set) var storage: Storage
 
-    @available(*, unavailable)
-    public var wrappedValue: Value {
-        get { fatalError("only works on instance properties of classes") }
-        set { fatalError("only works on instance properties of classes") }
+    public var wrappedValue: Bool {
+        get {
+            storage.value
+        }
+        nonmutating set {
+            storage.value = newValue
+            didSet?(newValue)
+            publisher.send(newValue)
+        }
     }
 
     public var projectedValue: Self {
         self
     }
 
-    public static subscript<EnclosingSelf: ObservableObject>(
-        _enclosingInstance object: EnclosingSelf,
-        wrapped wrappedKeyPath: ReferenceWritableKeyPath<EnclosingSelf, Value>,
-        storage storageKeyPath: ReferenceWritableKeyPath<EnclosingSelf, Self>
-    ) -> Value {
-        get {
-            let propertyWrapper = object[keyPath: storageKeyPath]
-            if let key = propertyWrapper.key {
-                return propertyWrapper.storage.object(forKey: key) as? Value ?? propertyWrapper.defaultValue
-            } else {
-                return propertyWrapper.value
+    public var action: DebugToggleAction {
+        DebugToggleAction(title: displayTitle, toggle: binding, defaultValue: defaultValue, publisher: publisher)
+    }
+
+    public var binding: Binding<Bool> {
+        Binding(
+            get: { wrappedValue },
+            set: { wrappedValue = $0 }
+        )
+    }
+
+    public init(
+        wrappedValue defaultValue: Bool,
+        title: String? = nil,
+        key: String,
+        storage: UserDefaults = .standard,
+        didSet: ((Bool) -> Void)? = nil
+    ) {
+        self.defaultValue = defaultValue
+        self.displayTitle = title ?? key.camelCaseToWords().replacingOccurrences(of: "Key", with: "").trimmingCharacters(in: .whitespaces)
+        self.publisher = PassthroughSubject<Bool, Never>()
+        self.storage = Storage(defaultValue, key: key, storage: storage)
+        self.didSet = didSet
+    }
+
+    public init(
+        wrappedValue defaultValue: Bool,
+        title: String,
+        didSet: ((Bool) -> Void)? = nil
+    ) {
+        self.defaultValue = defaultValue
+        self.displayTitle = title
+        self.publisher = PassthroughSubject<Bool, Never>()
+        self.storage = Storage(defaultValue)
+        self.didSet = didSet
+    }
+
+    final public class Storage: NSObject, ObservableObject {
+        let key: String?
+        var storedValue: Bool
+        var store: UserDefaults
+
+        var value: Bool {
+            get {
+                if let key {
```

Comment:
Will this compile on versions of Swift < 5.7... are we effectively dropping support for pre XCode 14/iOS 16 projects with this PR?

Reply:
Yeah great catch, that was definitely not my intention. This new if let is the only new Swift feature that causes builds to fail pre XCode 14. I'll revert to old style.